theano/gof/toolbox.py

```diff
@@ -100,8 +103,15 @@ def __init__(self, old_r, new_r, old_r_val, new_r_val, reason,
         self.reason = reason
         self.old_graph = old_graph
         self.new_graph = new_graph
+        self.full_err = None
+        if isinstance(old_r, str):
+            assert (new_r is None and old_r_val is None and new_r_val is None and
+                    reason is None and old_graph is None and new_graph is None)
+            self.full_err = old_r
```

Comment:
It might be confusing, but otherwise we end up with even more confusing errors like this:

```
TypeError: __init__ has 7 arguments but got only 1
```

Instead of the real error which just triggered a BadOptimization.